larray_editor/editor.py

```diff
@@ -554,6 +586,177 @@ def new(self):
             self.unsaved_modifications = False
             self.statusBar().showMessage("Viewer has been reset", 4000)
 
+    ##################################
+    #  METHODS TO SAVE/LOAD SCRIPTS  #
+    ##################################
+
+    def _load_script(self, filepath, lines, symbols, include_user_namespace=False):
+        # %load -r <lines> -s <symbols> -y -n source
+        if qtconsole_available:
+            try:
+                cmd = []
+                if lines:
+                    cmd += ['-r {}'.format(lines)]
+                if symbols:
+                    cmd += ['-s {}'.format(symbols)]
+                cmd += [filepath]
+                self.kernel.shell.run_line_magic('load', ' '.join(cmd))
+                self.ipython_cell_executed()
+                self.update_recent_script_list(filepath)
+                if include_user_namespace:
+                    pass
+            except Exception as e:
+                QMessageBox.critical(self, "Error", "Cannot load script file {}:\n{}"
+                                     .format(os.path.basename(filepath), e))
+        else:
+            QMessageBox.critical(self, "Error", "Cannot load script file {}. You need to install ipython and "\
+                                                "qtconsole.".format(os.path.basename(filepath)))
+
+    def load_script(self, filepath=None):
+        if self._ask_to_save_if_unsaved_modifications():
+            # %save add automatically the extension .py if not present in passed filename
+            dialog = QDialog(self)
+            layout = QGridLayout()
+            # filepath
+            browse_label = QLabel("Filepath")
+            browse_edit = QLineEdit()
+            browse_button = QPushButton("Browse")
+            if isinstance(filepath, str):
+                browse_edit.setText(filepath)
+            browse_filedialog = QFileDialog(self, filter="Python Script (*.py)")
+            browse_filedialog.setFileMode(QFileDialog.ExistingFile)
+            browse_button.clicked.connect(browse_filedialog.open)
+            browse_filedialog.fileSelected.connect(browse_edit.setText)
+            layout.addWidget(browse_label, 0, 0)
+            layout.addWidget(browse_edit, 0, 1)
+            layout.addWidget(browse_button, 0, 2)
+            # lines
+            lines_label = QLabel("Lines")
```

Comment:
I would rather have the lines_edit and symbols_edit "guarded" by a radio button with "all lines", "lines [line_edit]", and "variables [symbols_edit]" in the same vein than the pages selection in the print dialog of firefox or acrobat reader.
We would loose the possibility to load specific symbols within specific lines, but I guess that would be very rarely used.